src/materials/VelocityGaussianHeatSource.C

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************/
+/*                        DO NOT MODIFY THIS HEADER                         */
+/*                                                                          */
+/* MALAMUTE: MOOSE Application Library for Advanced Manufacturing UTilitiEs */
+/*                                                                          */
+/*           Copyright 2021 - 2023, Battelle Energy Alliance, LLC           */
+/*                           ALL RIGHTS RESERVED                            */
+/****************************************************************************/
+
+#include "VelocityGaussianHeatSource.h"
+
+registerMooseObject("MalamuteApp", VelocityGaussianHeatSource);
+
+InputParameters
+VelocityGaussianHeatSource::validParams()
+{
+  InputParameters params = GaussianHeatSourceBase::validParams();
+  params.addParam<Real>("x0", 0, "The x component of the initial center of the heating spot");
+  params.addParam<Real>("y0", 0, "The y component of the initial center of the heating spot");
+  params.addParam<Real>("z0", 0, "The z component of the initial center of the heating spot");
+
+  params.addParam<FunctionName>(
+      "function_vx",
+      0,
```

Comment:
should be the default function name "0"?

Reply:
I'll change the default to 'vx', 'vy' and 'vz'.

Reply:
^ no to above. Both "0" and 0 should work. I'll change to "0"